test/e2e/framework.go

```diff
@@ -698,6 +705,44 @@ func (data *TestData) deployAntreaFlowExporter(ipfixCollector string) error {
 	return data.mutateAntreaConfigMap(nil, ac, false, true)
 }
 
+// deployFlowVisibilityClickHouse deploys ClickHouse operator and DB.
+func (data *TestData) deployFlowVisibilityClickHouse() (string, error) {
+	rc, _, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s", clickHouseOperatorYMLUrl))
+	if err != nil || rc != 0 {
+		return "", fmt.Errorf("error when deploying ClickHouse operator: %v", err)
+	}
+
+	err = data.createNamespace(flowVisibilityNamespace, nil)
+	if err != nil {
+		return "", err
+	}
+	err = data.createNamespace(flowAggregatorNamespace, nil)
+	if err != nil {
+		return "", err
+	}
+	rc, _, _, err = provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s", flowVisibilityCHYML))
+	if err != nil || rc != 0 {
+		return "", fmt.Errorf("error when deploying the ClickHouse %s: %v", flowVisibilityCHYML, err)
+	}
+	clickHouseIP, err := testData.podWaitForIPs(defaultTimeout, "chi-clickhouse-clickhouse-0-0-0", flowVisibilityNamespace)
```

Comment:
Another observation in my local environment... The `chi-clickhouse-clickhouse-0-0-0` Pod won't be created immediately after we deploy the YAML. In the worst case, it takes 5-10 seconds until we can find this resource in the namespace. That's why I let it [sleep 10](https://github.com/antrea-io/antrea/blob/76386be40bd0cf8752779100920fbc4ec3ab338e/build/yamls/flow-visibility/start-flow-visibility.sh#L28) before looking up the Pod, in the deployment script. I assume it is because the resource `kind: "ClickHouseInstallation"` has some internal operations to create the Pod and Services, which takes some time. @dreamtalen can also confirm the issue in his local env. 
It seems like the method `podWaitForIPs` will report error if the resource is not found. Have you ever seen the error   while you test locally?

Reply:
podWaitFor series of funcs wait for target pod to be found first, then check for condition match. So no need for additional manual sleep before doing the polling wait. 

Switched to checking for pod readiness rather than IP in the latest change as it turns out CH pod does take a little while to be ready, which could unexpected break test in some cases.